mindsdb_native/libs/helpers/text_helpers.py

```diff
@@ -175,6 +186,29 @@ def is_foreign_key(data, column_name, data_subtype, other_potential_subtypes):
         all_uuid_charset = all(set(str(x)).issubset(uuid_charset) for x in data)
         is_uuid = all_uuid_charset and all_same_length
 
+        if all_same_length:
+            str_data = [str(x) for x in data]
+            
+            randomness_per_index = []
+            for i, _ in enumerate(str_data[0]):
+                all_ascii = all(isascii(x[i]) for x in str_data)
+                all_alpha = all_ascii and all(x[i].isalpha() for x in str_data)
+                all_numeric = all_ascii and all(x[i].isnumeric() for x in str_data)
+
+                if all_alpha:
+                    N = 26
+                elif all_numeric:
+                    N = 10
+
```

Comment:
The problematic column in the `automobile_insurance` dataset is actually a mix of alphanumerical characters, so ideally we'd want to check entropy in that case as well.

Reply:
What I mean is that the N should be 36 in that case, if I'm correct.

Reply:
actually this would work for `automobile_insurance` dataset (because it has only alphabetical characters in first 2 indices and others are numeric), but I changed computation for N to `len(set(x[i] for x in str_data))` because it's not restricted to any charset